src/prediction.jl

```diff
@@ -1,35 +1,68 @@
-non_na_mean(vv) = mean(vv[.!(isnan.(vv))])
+########################################
+# 0. Hierarchical Structures and Types #
+########################################
+
+abstract type Prediction end
+struct Classification <: Prediction end
+struct Regression <: Prediction end
+
+abstract type ResonancePredictor end
+abstract type ResonanceUnivariatePredictor <: ResonancePredictor end
+abstract type ResonanceMultivariatePredictor <: ResonancePredictor end
+
+mutable struct UnivariateRandomForestClassifier <: ResonanceUnivariatePredictor
+    name::String
+    inputs_outputs::Tuple{DataFrame, CategoricalArrays.CategoricalVector{Bool, UInt32, Bool, CategoricalArrays.CategoricalValue{Bool, UInt32}, Union{}}}
+    n_splits::Int64
+    dataset_partitions::Vector{Tuple{Vector{Int64}, Vector{Int64}}}
+    models::Vector{Machine}
+    selected_split::Tuple{Float64, Int64}
+    train_accuracies::Vector{Float64}
+    test_accuracies::Vector{Float64}
+end
 
-function fix_metadata_colnames!(longdata_df::DataFrame)
+mutable struct UnivariateRandomForestRegressor <: ResonanceUnivariatePredictor
+    name::String
+    inputs_outputs::Tuple{DataFrame, Vector{Float64}}
+    n_splits::Int64
+    dataset_partitions::Vector{Tuple{Vector{Int64}, Vector{Int64}}}
+    models::Vector{Machine}
+    slope_corrections::Vector{}
+    selected_split::Tuple{Float64, Int64}
+    train_maes::Vector{Float64}
+    test_maes::Vector{Float64}
+    train_mapes::Vector{Float64}
+    test_mapes::Vector{Float64}
+    train_cor::Vector{Float64}
+    test_cor::Vector{Float64}
+end
 
-    ## This function was created to fix small inconsistencies and typos on the longdata table.
-    ## As original data cleanup and wrangling progresses, it will be altered and ultimately deprecated.
-    ## Comments are provided to help review transformations
+mutable struct UnivariatePredictorEnsemble
+    screen_names::Vector{String}
+    col_names::Vector{Symbol}
+    predictors::Vector{T where T <: ResonanceUnivariatePredictor}
+end
 
-    # 1. Rename "[Collinsella]_massiliensis" to "Collinsella_massiliensis"
-    longdata_df[longdata_df.variable .== "[Collinsella]_massiliensis", :variable] .= "Collinsella_massiliensis"
+########################################
+# 1. Preprocessing/Wrangling Functions #
+########################################
 
-    return(longdata_df)
-    
+Resonance.dropmissing(vv) = vv[.!(ismissing.(vv))]
+dropnan(vv) = vv[.!(isnan.(vv))]
+nonna_mean(vv) = mean(dropnan(vv))
+nonmissing_mean(vv) = mean(dropmissing(vv))
+nonna_nonmissing_mean(vv) = mean(dropnan(dropmissing(vv)))
```

Comment:
Not a fan of this. It seems like it doesn't save a lot of typing to just do this at the call site. If it were doing something special on a type that we're introducing, fine, but `nonna_nonmissing_mean` vs `mean(dropnan(dropmissing(` doesn't add much IMO.

Reply:
Resolved.